tsl/src/continuous_aggs/materialize.c

```diff
@@ -298,31 +292,19 @@ get_materialization_end_point_for_table(int32 raw_hypertable_id, int32 materiali
 	Dimension *time_column = hyperspace_get_open_dimension(raw_table->space, 0);
 	NameData time_column_name = time_column->fd.column_name;
 	Oid time_column_type = ts_dimension_get_partition_type(time_column);
-	bool found_new_tuples = false;
+	int64 now_time = ts_get_now_internal(time_column);
+	int64 end_time, start_time;
 
-	found_new_tuples = hypertable_get_min_and_max(hypertable,
-												  &time_column_name,
-												  old_completed_threshold,
-												  time_column_type,
-												  &start_time,
-												  &end_time);
-
-	if (!found_new_tuples)
+	start_time = old_completed_threshold;
+	if (start_time == PG_INT64_MIN)
 	{
-		if (verbose)
-			elog(INFO,
-				 "new materialization range not found for %s.%s (time column %s): no new data",
-				 NameStr(*hypertable.schema),
-				 NameStr(*hypertable.name),
-				 NameStr(time_column_name));
-		*materializing_new_range = false;
-		return old_completed_threshold;
+		/* If there is no completion threshold yet set, find the minimum value stored in the
+		 * hypertable */
+		start_time = hypertable_get_min(hypertable, &time_column_name, time_column_type);
```

Comment:
this function call seems to be unconditional, unlike the comment?

Reply:
huh? line 202 is `if (start_time == PG_INT64_MIN)`